tf_yarn/_dispatch_task.py

```diff
@@ -38,9 +38,11 @@
 
 def main(
     experiment_fn: ExperimentFn,
-    all_tasks: typing.List[str]
+    all_tasks: typing.List[str],
+    num_threads : int = 0
```

Comment:
Maybe we should choose another default and compute this value adapted to our infra ? 0 leaves the choice to tf and we don't know what happens.

This setting actually depends on the vcores we specify for each TaskSpec. If we ask for 4 vcores we have 1 real CPU available and therefore we could have 1 thread, for 8 vcores we could afford 2 threads.

We could copy what does Spark which defines:
spark.task.cpus => the number of cores to allocate for each task, 1 by default
spark.executor.cores => number of tasks and also asked by yarn for our CPU budget allocation
=> the number of real tasks/threads by executor is vcores/spark.task.cpus

Maybe we should do the same ? 
vcores is currently specified in the TaskSpec therefore unique for each worker. We could add a setting task.cpus specific for each worker 
thread_num = vcores/task.cpus

On Criteo infra the quota of vCores to real CPUs is 4 to 1. 
So we would specify vcores = 4, task.cpus = 4 => threads = 1
Not perfect. But we know it works in Spark and would be consistent.

Reply:
Yes, I just added a num_thread as an argument to the run_on_yarn() function so the user would specify his/her own num_threads in the tf code --run_on_yarn(...,num_threads=num_threads)... It looks like 0 is the default setting and the current version of tf-yarn uses it if num_threads is not set already in the code. I have committed this new version but we can also make it adaptive or find a rule as you mentioned and then the user doesn't need to specify it by hand in the tf code

Reply:
I also removed num_threads=0 from main as default--it was supposed to be in run_on_yarn actually, otherwise the system always put it 0. According to TF doc--0 means the system picks an appropriate number.